applications/ConvectionDiffusionApplication/python_scripts/coupled_structural_thermal_solver.py

```diff
@@ -0,0 +1,164 @@
+        if self.domain_size == 2:
+            modeler.GenerateModelPart(self.structural_solver.main_model_part,
+                                      self.thermal_solver.main_model_part,
+                                      "EulerianConvDiff2D",
+                                      "ThermalFace2D2N")
+        else:
+            modeler.GenerateModelPart(self.structural_solver.main_model_part,
+                                      self.thermal_solver.main_model_part,
+                                      "EulerianConvDiff3D",
+                                      "ThermalFace3D3N")
+
+        # Set the saved convection diffusion settings to the new thermal model part
+        self.thermal_solver.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.CONVECTION_DIFFUSION_SETTINGS, convection_diffusion_settings)
+
+    def PrepareModelPart(self):
+        self.structural_solver.PrepareModelPart()
+        self.thermal_solver.PrepareModelPart()
+
+    def AddDofs(self):
+        self.structural_solver.AddDofs()
+        self.thermal_solver.AddDofs()
+
+    def AdaptMesh(self):
+        pass
+
+    def GetComputingModelPart(self):
+        return self.structural_solver.GetComputingModelPart()
+
+    def GetOutputVariables(self):
+        pass
+
+    def ComputeDeltaTime(self):
+        return self.structural_solver._ComputeDeltaTime()
+
+    def GetMinimumBufferSize(self):
+        buffer_size_fluid = self.structural_solver.GetMinimumBufferSize()
+        buffer_size_thermal = self.thermal_solver.GetMinimumBufferSize()
+        return max(buffer_size_fluid, buffer_size_thermal)
+
+    def Initialize(self):
+        self.structural_solver.Initialize()
+        self.thermal_solver.Initialize()
+
+    def Clear(self):
+        (self.structural_solver).Clear()
+        (self.thermal_solver).Clear()
+
+    def Check(self):
+        (self.structural_solver).Check()
+        (self.thermal_solver).Check()
+
+    def SetEchoLevel(self, level):
+        (self.structural_solver).SetEchoLevel(level)
+        (self.thermal_solver).SetEchoLevel(level)
+
+    def AdvanceInTime(self, current_time):
+        #NOTE: the cloning is done ONLY ONCE since the nodes are shared
+        new_time = self.structural_solver.AdvanceInTime(current_time)
+        return new_time
+
+    def InitializeSolutionStep(self):
+        self.structural_solver.InitializeSolutionStep()
+        self.thermal_solver.InitializeSolutionStep()
+
+    def Predict(self):
+        self.structural_solver.Predict()
+        self.thermal_solver.Predict()
+
+    def SolveSolutionStep(self):
+        fluid_is_converged = self.structural_solver.SolveSolutionStep()
+        thermal_is_converged = self.thermal_solver.SolveSolutionStep()
+
+        return (fluid_is_converged and thermal_is_converged)
+
+    def FinalizeSolutionStep(self):
+        self.structural_solver.FinalizeSolutionStep()
+        self.thermal_solver.FinalizeSolutionStep()
+
+    def Solve(self):
```

Comment:
Same, there was an agreement to call the functions separately

Reply:
Agree. It can be removed.

Reply:
Removed